Ruby Bubble Sort, Project # 1, Ruby - Request for Code review

wickathou opened the pull request:
Features:

A sorting algorithm built with Ruby

wickathou committed:
Added rb file

Grifo89 committed:
bubble_sort function done

wickathou committed:
Added bubble_sort_by

wickathou committed:
Code working on bubble sor and sort by

wickathou committed:
Linter added

wickathou committed:
Readme added

wickathou committed:
Readme added

Grifo89 committed:
fixed linter errors

Grifo89 committed:
Rubocop fixes

Ispirett commented on bubble_sort.rb:
```diff
@@ -0,0 +1,37 @@
+    end
+    ite += 1
+  end
+  puts arr
```
Please remove puts and just return the array

wickathou replied:
Done

wickathou replied:
Done

Ispirett commented on bubble_sort.rb:
```diff
@@ -0,0 +1,37 @@
+    end
+    ite += 1
+  end
+  puts arr
```
Please remove puts and just return the array.

wickathou replied:
Done

Ispirett reviewed (changes requested):
## Status: Changes Required
Excellent Work!! @wickathou  @Grifo89  with this project so far :+1:  :100: 

![image](https://user-images.githubusercontent.com/38801423/62908290-41739e00-bd45-11e9-922f-6b7cf52979ee.png)

Currently, your project returns nil 
 - [x] Checkout comments below to solve this issue.

 - [x] Please add stickler to your project, follow the instructions from pathwright.

**please rememeber to us more descriptive commit messages.**

wickathou commented:
We had to use Rubocop as we were having issues with Stickler.

wickathou committed:
Changes:
+ Updated line 16 and 32, from puts to return
+ Stickler not used, as we had issues with it. We used Rubocop instead

wickathou committed:
Changes:
+ Updated line 16 and 32, from puts to return
+ Stickler not used, as we had issues with it. We used Rubocop instead

wathigo commented on bubble_sort.rb:
```diff
@@ -0,0 +1,33 @@
+        arr[n] = arr[n + 1]
+        arr[n + 1] = key
+      end
+    end
```
**[REQUIRED]** 
We are missing the `end` statement for closing the `while` block.

wathigo commented on bubble_sort.rb:
```diff
@@ -0,0 +1,33 @@
+      if yield(arr[n], arr[n + 1]).positive?
+        arr[n], arr[n + 1] = arr[n + 1], arr[n]
+      end
+    end
```
**[REQUIRED]** 
We are missing the `end` statement for closing the `while` block.

wathigo commented on stickler.yml:
```diff
@@ -0,0 +1,1270 @@
+
```
**[REQUIRED]**
Let's the right configurations to the `.stickler.yml` file. Lets remove the html mark up from this file.  
Check out the correct linter configurations for Ruby projects [here](https://github.com/microverseinc/linters-config/tree/master/ruby).

wickathou replied:
That's the one we used and it's not working for us. Rubocop is allowed as it's described on the pathwright.

wathigo replied:
Let's remember to name the file correctly. It should be `.stickler.yml` **not** `stickler.yml`

wathigo commented on rubocop.yml:
```diff
@@ -0,0 +1,1385 @@
+
```
**[REQUIRED]**
Let's include the right linter configurations in `.rubocop.yml` file.
These are the right configs.
```
AllCops:
  Exclude:
    - "bin/*"
    - "db/**/*"
    - "config/**/*"
    - "Guardfile"
    - "Rakefile"
  DisplayCopNames: true

Metrics/LineLength:
  Max: 120
Metrics/MethodLength:
  Include: ["app/controllers/*"]
  Max: 20
Metrics/AbcSize:
  Include: ["app/controllers/*"]
  Max: 30
Metrics/ClassLength:
  Max: 150
Metrics/BlockLength:
  ExcludedMethods: ['describe']

Style/Documentation:
  Enabled: false
Style/ClassAndModuleChildren:
  Enabled: false

Layout/AlignHash:
  EnforcedColonStyle: key
Layout/ExtraSpacing:
  AllowForAlignment: false
Layout/MultilineMethodCallIndentation:
  Enabled: true
  EnforcedStyle: indented
```
Checkout this [repo](https://github.com/microverseinc/linters-config/blob/master/ruby/.rubocop.yml) for the right configurations for this file.

wickathou replied:
We used directly the ones in this repo
https://github.com/microverseinc/linters-config/tree/master/ruby

wathigo reviewed (changes requested):
### Hi @wickathou  and @Grifo89  :wave:  
Great work implementing the requested changes but we need to fix a few issues before we move on.* 

* See the issues pointed out in the comments below :point_down:  

**After implementing the changes, submit another request for review.**

_Have a great weekend :smile:  
Reviewed by: **Simon Wathigo**_

wickathou committed:
updated stickler, rubocop and added end on the while statements

Grifo89 committed:
Rubocop fixes

Grifo89 committed:
fixed files name

Grifo89 committed:
fixed files name

Grifo89 committed:
Added gitignore file

Grifo89 committed:
fixed stickler error

Grifo89 commented:
Hey there, changes and corrections are made.

Grifo89 commented:
[](url)

samgaco commented on bubble_sort.rb:
```diff
@@ -0,0 +1,33 @@
+      end
+    end
+    arr
+  end
```
:+1:

samgaco reviewed (approved):
Everything is fine, you may move on :+1: 

------------------------------------------

* Friendly reminder: You should use [Github flow](https://microverse.pathwright.com/library/microverse-onboarding/86777/path/step/53987922/) in this project. However, changing it now would take too much time. Please make sure you use [Github Flow](https://microverse.pathwright.com/library/microverse-onboarding/86777/path/step/53987922/) in the next project. 
------------------------------------------

* ###### STATUS OF THE PROJECT: APPROVED :heavy_check_mark:
